### ISSUE TITLE
Need android devices on tryserver.client.catapult

### ISSUE THREAD

**Issue by anniesullie** [catapult-project/catapult#2403]
The hardware was set up in crbug.com/505859 but we still need our recipe to utilize these bots.

Code locations:
Our recipe: https://cs.chromium.org/chromium/build/scripts/slave/recipes/catapult.py
Catapult-side steps: https://github.com/catapult-project/catapult/blob/master/catapult_build/build_steps.py
Master config: https://cs.chromium.org/chromium/build/masters/master.tryserver.client.catapult/builders.pyl

**Comment by jbudorick** [comment]
Bot's up: https://build.chromium.org/p/tryserver.client.catapult/builders/Catapult%20Android%20Tryserver/builds/0

Going to enable telemetry_perf_unittests and possibly telemetry_unittests tomorrow.

**Comment by jbudorick** [comment]
Do you want a Catapult Android bot on client.catapult as well?

**Comment by anniesullie** [comment]
I think it's okay to just have them on the tryservers. Thanks so much for your help with this!

**Comment by anniesullie** [comment]
Oh we already have two? Maybe we could speed up the CQ by putting both bots on the tryservers.

**Comment by jbudorick** [comment]
Yeah. I've got them both set up on the tryserver now, but build1-b3 is down.

**Comment by jbudorick** [comment]
@anniesullie @nedn are there other suites that'd make sense on the Android bot? I'm not familiar with most of the catapult suites beyond telemetry, devil, and devil's dependencies.

**Comment by anniesullie** [comment]
@zeptonaut @randalnephew @Alex319 are there BattOr tests we would want to run on these devices? (We'd have to get BattOrs in before it makes sense though)

Long-term I think we'd want to run trace-viewer UI tests, but in the medium term we're not ready to actually support trace-viewer mobile UI.

**Comment by nedn** [comment]
systrace is a good one to run again real devices.
@ChrisCraik

**Comment by anniesullie** [comment]
Also @wangzhen127 re systrace tests on android catapult cq.

**Comment by zeptonaut** [comment]
Realistically, I think that making trace viewer usable on Android usable would be a huge enough task that it'll get pushed off forever. 

Yes, though, we definitely want BattOr/Android integration tests on the CQ when we get more BattOrs in.

**Comment by wangzhen127** [comment]
Nice! Have been waiting for this. Thanks!

**Comment by jbudorick** [comment]
Looping back to this -- are there systrace tests that run on a device? I see systrace/bin/run_tests, but that appears to run on the host machine.

@wangzhen127 @ChrisCraik

**Comment by wangzhen127** [comment]
No. Currently the systrace tests are running just on the host machine. My intern will add more Android tests when merging systrace and adb_profile_chrome this summer.

**Comment by wangzhen127** [comment]
@peterwashington

**Comment by peterwashington** [comment]
Yup, I plan to add Android tests for the merged systrace and adb_profile_chrome, so I will modify catapult/catapult_build/build_steps.py to point to a new set of tests that will run exclusively for Android (in addition to the current set of systrace tests that currently do not run on Android).  

Let me know if there is anything special that I need to keep in mind when writing the Android systrace tests.

**Comment by zeptonaut** [comment]
@anniesullie, in response to your earlier question, I definitely think that we want [this sanity integration test](https://cs.corp.google.com/github/catapult-project/catapult/common/battor/battor/battor_wrapper_devicetest.py?type=cs&q=p:github+f:%5Ecatapult-project/catapult/+battor+f:test&l=79) (which is currently running on the Windows tryserver) running on all platforms, including Android.

**Comment by peterwashington** [comment]
@anniesullie @zeptonaut @nedn @jbudorick 

A couple of questions about the current Android Trybots:
1) Is the Android device rooted?
2) Is atrace installed?
3) What version of Chromium is installed?
4) What Android version is running on the device(s)?

I ask because the profile_chrome unit tests work fine on my local machine (Ubuntu 4.8.4-2ubuntu1~14.04.3) with a rooted Android device (Nexus 5) with Android version 6.0. When I run the tests on the Android Trybot, however, I get a bunch of errors which may or may not be caused by an Android Trybot configuration that is incompatible with the profile_chrome tests: https://build.chromium.org/p/tryserver.client.catapult/builders/Catapult%20Android%20Tryserver/builds/805/steps/Systrace%20Tests/logs/stdio

**Comment by anniesullie** [comment]
You can answer some of the questions by looking at the Android Device Status logs: https://build.chromium.org/p/tryserver.client.catapult/builders/Catapult%20Android%20Tryserver/builds/805/steps/Android%3A%20Device%20Status/logs/stdio

The device is rooted and using K:
```
OS build fingerprint: google/hammerhead/hammerhead:4.4.4/KTU84P/1227136:userdebug/dev-keys
```

@jbudorick @randalnephew do you know how to tell which version of chromium is installed and whether atrace is installed?

**Comment by jbudorick** [comment]
I'm not sure about atrace. @randalnephew or @Alex319?

We don't currently report the installed system version of chrome anywhere, and AFAICT none of the current suites are installing other versions of chrome or chromium. What would you like to run the systrace tests against?

**Comment by peterwashington** [comment]
@Alex319 @randalnephew Any news about whether the Android Trybots have atrace installed?

@jbudorick I believe that any recent version of Chrome or Chromium will do, but the unit tests run on "stable". Probably no need to install a different version of the browser but if I knew the version, I could install it locally and see if the version of the browser is the culprit (which I doubt but it is good to see the options). 

Also, I installed Android K locally and I am not getting any bugs, so this is not the source of the problem. 

Still debugging to determine potential other sources of the problem.
